.github/workflows/pr-mm.yaml

```diff
@@ -0,0 +1,176 @@
+          echo " -- update the package cache"
+          sudo apt update
+          echo " -- upgradables"
+          sudo apt list --upgradable
+          echo " -- upgrade"
+          sudo apt dist-upgrade
+          echo " -- install our dependencies"
+          sudo apt install -y make openssh-server libgsl-dev libopenmpi-dev libhdf5-dev
+
+      - name: python ${{matrix.python}} setup
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{matrix.python}}
+
+      - name: install dependencies
+        run: |
+          python${pythonVersion} -m pip install --upgrade pip
+          pip3 install distro graphene numpy pybind11 PyYAML
+        env:
+          pythonVersion: ${{matrix.python}}
+
+      - name: list system packages
+        run: |
+          dpkg -l
+
+      - name: versions
+        run: |
+          echo " -- platform tag from python"
+          python${pythonVersion} -c "import sys; print(sys.platform)"
+          echo " -- distribution tag from python"
+          python${pythonVersion} -c "import distro; print(distro.linux_distribution(full_distribution_name=False))"
+          echo " -- make"
+          make --version
+          echo " -- python"
+          python${pythonVersion} --version
+          echo "    -- prefix"
+          python${pythonVersion}-config --prefix
+          echo "    -- includes"
+          python${pythonVersion}-config --includes
+          echo "    -- libs"
+          python${pythonVersion}-config --libs
+          echo " -- ${{matrix.cc}}"
+          ${{matrix.cc}} --version
+          echo " -- ${{matrix.cxx}}"
+          ${{matrix.cxx}} --version
+        env:
+          pythonVersion: ${{matrix.python}}
+
+      - name: external dependencies
+        run: |
+          echo " -- gsl"
+          sudo dpkg-query -L libgsl-dev
+          echo " -- openmpi"
+          sudo dpkg-query -L libopenmpi-dev
+          echo " -- hdf5"
+          sudo dpkg-query -L libhdf5-dev
+
+      - name: locations of python packages
+        run: |
+          echo " -- distro"
+          pip3 show distro
+          echo " -- graphene"
+          pip3 show graphene
+          echo " -- numpy"
+          pip3 show numpy
+          echo " -- its headers"
+          find ${pythonLocation}/lib/python${pythonVersion}/site-packages/numpy/core/include
+          echo " -- pybind11"
+          pip3 show pybind11
+          echo " -- its headers"
+          find ${pythonLocation}/lib/python${pythonVersion}/site-packages/pybind11
+          echo " -- yaml"
+          pip3 show PyYAML
+        env:
+          pythonVersion: ${{matrix.python}}
+
+      - name: clone mm
+        run: |
+          echo " -- cloning mm"
+          git clone https://github.com/aivazis/mm
+
+      - name: checkout pyre
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          path: 'pyre'
+
+      - name: mm diagnostics
+        run: |
+          echo " -- switching to the pyre home directory"
```

Comment:
`mm` was launched outside the `pyre` source directory on purpose to exercise the tool itself without relying on a correct project configuration file. This way, if `mm` itself fails while attempting to build `pyre`, I know it is a  problem with the project configuration, not the tool